routing.py

```diff
@@ -1,19 +1,39 @@
 import logging
 import urllib.parse as urlparse
+from enum import Enum
 from typing import Tuple
 
 from domain.metric import MetricType
 from domain.types import AgentID
 
-MAIN = "/"
-MATRIX = "/matrix"
-TIME_SERIES = "/time-series"
-
 logger = logging.getLogger("routing")
 
 
+class Route(Enum):
+    INDEX = "/"
+    MATRIX = "/matrix"
+    TIME_SERIES = "/time-series"
+    UNKNOWN = "[unknown_route]"
+
+
+def extract_route(pathname: str) -> Route:
+    """
+    Example:
+        pathname: /matrix?metric=Latency
+        return:   Route.MATRIX
+    """
+
+    route_end_index = pathname.find("?")
+    route_str = pathname[:route_end_index] if route_end_index != -1 else pathname
+    try:
+        return Route(route_str)
+    except Exception:
```

Comment:
_Non-critical:_

The `except` is unnecessarily broad. It should be enough to catch `ValueError`.